src/main/java/de/tum/in/www1/artemis/domain/BaseExercise.java

```diff
@@ -236,6 +237,20 @@ private static boolean isValidAssessmentDueDate(ZonedDateTime releaseDate, Zoned
         return isBeforeAndNotNull(dueDate, assessmentDueDate) && isBeforeAndNotNull(releaseDate, assessmentDueDate);
     }
 
+    /**
+     * This method is used to validate the exampleSolutionPublicationDate of an exercise. An exampleSolutionPublicationDate is valid if it is after the releaseDate and dueDate.
+     * Any given exampleSolutionPublicationDate is valid if releaseDate and dueDate are not set.
+     * exampleSolutionPublicationDate is valid if it is not set.
```

Comment:
You could think about adding `@param` to your comment. 
And the following method could be:
`exampleSolutionPublicationDate == null || (isBeforeAndNotNull(dueDate, exampleSolutionPublicationDate) && isBeforeAndNotNull(releaseDate, exampleSolutionPublicationDate));`. 

But this is only a suggestion. In my opinion, both ways are fine.

Reply:
Thanks for the suggestions Dominik! I would prefer to keep both as it is right now due to being consistent with the rest of the files. `@param` is not used for the same date type parameters in other validation methods, probably because it would just repeat the name of the date parameter.

And null checks are handled with a return early pattern in the other validation methods as well which enables the reader capture the date constraints with a quick glance in my opinion.

Reply:
Good and valid points :+1: thank you for your explanation.